pkg/trimaran/resourcestats.go

```diff
@@ -136,3 +136,124 @@ func GetResourceRequested(pod *v1.Pod) *framework.Resource {
 	}
 	return result
 }
+
+// GetResourceLimits : calculate the resource limits of a pod (CPU and Memory)
+func GetResourceLimits(pod *v1.Pod) *framework.Resource {
+	// add up limit of all containers
+	result := &framework.Resource{}
+	for _, container := range pod.Spec.Containers {
+		result.Add(container.Resources.Limits)
+	}
+	// take max_resource(sum_pod, any_init_container)
+	for _, container := range pod.Spec.InitContainers {
+		for rName, rQuantity := range container.Resources.Limits {
+			switch rName {
+			case v1.ResourceCPU:
+				if cpu := rQuantity.MilliValue(); cpu > result.MilliCPU {
+					result.MilliCPU = cpu
+				}
+			case v1.ResourceMemory:
+				if mem := rQuantity.Value(); mem > result.Memory {
+					result.Memory = mem
+				}
+			default:
+			}
+		}
+	}
+	return result
+}
+
+// NodeRequestsAndLimits : data ralated to requests and limits of resources on a node
+type NodeRequestsAndLimits struct {
+	// NodeRequest sum of requests of all pods on node
+	NodeRequest *framework.Resource
+	// NodeLimit sum of limits of all pods on node
+	NodeLimit *framework.Resource
+	// NodeRequestMinusPod is the NodeRequest without the requests of the pending pod
+	NodeRequestMinusPod *framework.Resource
+	// NodeLimitMinusPod is the NodeLimit without the limits of the pending pod
+	NodeLimitMinusPod *framework.Resource
+	// Nodecapacity is the capacity (allocatable) of node
+	Nodecapacity *framework.Resource
+}
+
+// GetNodeRequestsAndLimits : total requested and limits of resources on a given node plus a pod
+func GetNodeRequestsAndLimits(podsOnNode []*v1.Pod, node *v1.Node, pod *v1.Pod) *NodeRequestsAndLimits {
+	// initialization
+	nodeRequest := &framework.Resource{}
+	nodeLimit := &framework.Resource{}
+	unlimitedCpu := false
+	unlimitedMem := false
+	nodeRequestMinusPod := &framework.Resource{}
+	nodeLimitMinusPod := &framework.Resource{}
+	// set capacities
+	nodeCapacity := &framework.Resource{}
+	allocatableResources := node.Status.Allocatable
+	amCpu := allocatableResources[v1.ResourceCPU]
```

Comment:
Other than cpu and memory, other core resources can also be overcommitted, like ephemeral storage.

Reply:
Yes, it would be nice to add support for additional resources such as GPUs and ephemeral storage. We'll hold to that until the load watcher supports exporting usage metrics for these resources.